File: sentimental-mario/mario.py
# TODO
# program that prints out pyramid with a height of the inputed value (0-8)
from cs50 import get_int


def main():

    height = get_height()

    counter = 0
    # do height times
    for i in range(height):
        counter += 1
        # add the whitespaces
        for i in range((height-1)-i):
            print(" ", end="")
        # add the hash symbol
        for i in range(counter):
            print("#", end="")
        # print new line
        print("\n", end="")


def get_height():
    # ask for height until user cooperates
    while True:
        try:
            n = get_int("Height:")
            if n > 0 and n <= 8:
                return n
        except ValueError:
            print("input an integer from 0 - 8")


main()

File: mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //input and height
    int x;
    //counter
    int i;
    //spaces
    int j;
    //hashes
    int k;

    //ask for height input x while x > 8
    do
    {
        x = get_int("Enter the height 1-8: ");
    }
    while (x > 8 || x <= 0);

    //do x times (height)
    for (i = 1; i < x + 1; i++)
    {

        //print x-1 spaces
        for (j = x - i; j > 0; j--)
        {
            printf(" ");
        }

        //print 0+1 hashes
        for (k = 0; k < i; k++)
        {
            printf("#");
        }

        //next line
        printf("\n");

    }
}
